azurecompute/src/main/java/org/jclouds/azurecompute/util/ConflictManagementPredicate.java

```diff
@@ -195,14 +199,11 @@ public boolean apply(final String input) {
          switch (operation.status()) {
             case SUCCEEDED:
                return true;
-
             case IN_PROGRESS:
             case UNRECOGNIZED:
                return false;
-
             case FAILED:
-               throw new RuntimeException(new CancellationException(input));
-
+               throw new RuntimeException(operation.error().message());
```

Comment:
I'd be inclined to be more defensive in this method. For example, if `api.getOperationApi().get(input)` returns null then we'd get an NPE. Similarly now, if operation.error() returns null we'd get an NPE (which hopefully wouldn't happen because we have status=FAILED, but I don't trust cloud provider APIs much!)